### ISSUE TITLE
C types - SDB properties

### ISSUE THREAD

**Issue by XVilka** [radare/radare2#7710]
- [ ] Add support for bitfields
 - [ ] Add support for alignment property of the type
 - [ ] Support for C++ objects

**Comment by radare** [comment]
C++?

**Comment by XVilka** [comment]
@radare yes, not sure how to do that, but I think we need to support more complex types, for languages like C++/D, Rust and Go, Haskell, etc.

**Comment by XVilka** [comment]
For this we need to think/discuss first how it should be done exactly @sivaramaaa

**Comment by XVilka** [comment]
@sivaramaaa for your information - in tcc parser and SDB database some metainformation about type stored in "meta" field, which can be enhanced, e.g. used as a bitfield to store more different properties (you can grep `shlr/tcc/tccgen.c` for "meta" keyword to see the code creating it).

**Comment by thestr4ng3r** [comment]
About C++ types, currently I am implementing the anal classes completely separate from types to be saved in sdb under the `anal/classes` namespace. The reason is that classes are quite independent of actual types in many cases (static methods, vtables, rtti, etc.).
The idea would be to later add references between types (structs) and classes to for example describe the memory layout of a class.
However I am open for better suggestions if anybody has a better idea.
